File: CS/HowToRemoveNewButton.Web/ApplicationCode/WebApplication.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using DevExpress.ExpressApp.Web;

namespace HowToRemoveNewButton.Web {
   public partial class HowToRemoveNewButtonAspNetApplication : WebApplication {
      private DevExpress.ExpressApp.SystemModule.SystemModule module1;
      private DevExpress.ExpressApp.Web.SystemModule.SystemAspNetModule module2;
      private HowToRemoveNewButton.Module.HowToRemoveNewButtonModule module3;
      private HowToRemoveNewButton.Module.Web.HowToRemoveNewButtonAspNetModule module4;
      private DevExpress.ExpressApp.Security.SecurityModule securityModule1;
      private DevExpress.ExpressApp.Security.SecuritySimple securitySimple1;
      private DevExpress.ExpressApp.Objects.BusinessClassLibraryCustomizationModule module6;
      private DevExpress.ExpressApp.Security.AuthenticationActiveDirectory authenticationActiveDirectory1;
      private System.Data.SqlClient.SqlConnection sqlConnection1;
      private DevExpress.ExpressApp.Validation.ValidationModule module5;

      public HowToRemoveNewButtonAspNetApplication() {
         InitializeComponent();
      }

      private void HowToRemoveNewButtonAspNetApplication_DatabaseVersionMismatch(object sender, DevExpress.ExpressApp.DatabaseVersionMismatchEventArgs e) {
         if (System.Diagnostics.Debugger.IsAttached)
         {
            e.Updater.Update();
            e.Handled = true;
         }
         else
         {
            throw new InvalidOperationException(
               "The application cannot connect to the specified database, because the latter doesn't exist or its version is older than that of the application.\r\n" +
               "The automatical update is disabled, because the application was started without debugging.\r\n" +
               "You should start the application under Visual Studio, or modify the " +
               "source code of the 'DatabaseVersionMismatch' event handler to enable automatic database update, " +
               "or manually create a database with the help of the 'DBUpdater' tool.");
         }
      }

      private void InitializeComponent() {
         this.module1 = new DevExpress.ExpressApp.SystemModule.SystemModule();
         this.module2 = new DevExpress.ExpressApp.Web.SystemModule.SystemAspNetModule();
         this.module3 = new HowToRemoveNewButton.Module.HowToRemoveNewButtonModule();
         this.module4 = new HowToRemoveNewButton.Module.Web.HowToRemoveNewButtonAspNetModule();
         this.module5 = new DevExpress.ExpressApp.Validation.ValidationModule();
         this.module6 = new DevExpress.ExpressApp.Objects.BusinessClassLibraryCustomizationModule();
         this.securityModule1 = new DevExpress.ExpressApp.Security.SecurityModule();
         this.securitySimple1 = new DevExpress.ExpressApp.Security.SecuritySimple();
         this.authenticationActiveDirectory1 = new DevExpress.ExpressApp.Security.AuthenticationActiveDirectory();
         this.sqlConnection1 = new System.Data.SqlClient.SqlConnection();
         ((System.ComponentModel.ISupportInitialize)(this)).BeginInit();
         // 
         // securitySimple1
         // 
         this.securitySimple1.Authentication = this.authenticationActiveDirectory1;
         this.securitySimple1.UserType = typeof(DevExpress.Persistent.BaseImpl.BasicUser);
         // 
         // authenticationActiveDirectory1
         // 
         this.authenticationActiveDirectory1.CreateUserAutomatically = true;
         this.authenticationActiveDirectory1.UserType = typeof(DevExpress.Persistent.BaseImpl.BasicUser);
         // 
         // sqlConnection1
         // 
         this.sqlConnection1.ConnectionString = "Data Source=(local);Initial Catalog=HowToRemoveNewButton;Integrated Security=SSPI;Pooling=false";
         this.sqlConnection1.FireInfoMessageEventOnUserErrors = false;
         // 
         // HowToRemoveNewButtonAspNetApplication
         // 
         this.ApplicationName = "HowToRemoveNewButton";
         this.Connection = this.sqlConnection1;
         this.Modules.Add(this.module1);
         this.Modules.Add(this.module2);
         this.Modules.Add(this.module3);
         this.Modules.Add(this.module4);
         this.Modules.Add(this.module5);
         this.Modules.Add(this.module6);

         this.Modules.Add(this.securityModule1);
         this.Security = this.securitySimple1;
         this.DatabaseVersionMismatch += new System.EventHandler<DevExpress.ExpressApp.DatabaseVersionMismatchEventArgs>(this.HowToRemoveNewButtonAspNetApplication_DatabaseVersionMismatch);
         ((System.ComponentModel.ISupportInitialize)(this)).EndInit();

      }
   }
}

File: VB/RemoveNewButtonInLookups.Web/ApplicationCode/WebApplication.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports DevExpress.ExpressApp.Web

Namespace RemoveNewButtonInLookups.Web
    Partial Public Class RemoveNewButtonInLookupsAspNetApplication
        Inherits WebApplication
        Private module1 As DevExpress.ExpressApp.SystemModule.SystemModule
        Private module2 As DevExpress.ExpressApp.Web.SystemModule.SystemAspNetModule
        Private module3 As RemoveNewButtonInLookups.Module.RemoveNewButtonInLookupsModule
        'private RemoveNewButtonInLookups.Module.Web.RemoveNewButtonInLookupsAspNetModule module4;
        Private securityModule1 As DevExpress.ExpressApp.Security.SecurityModule
        Private module6 As DevExpress.ExpressApp.Objects.BusinessClassLibraryCustomizationModule
        Private module5 As DevExpress.ExpressApp.Validation.ValidationModule

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub RemoveNewButtonInLookupsAspNetApplication_DatabaseVersionMismatch(ByVal sender As Object, ByVal e As DevExpress.ExpressApp.DatabaseVersionMismatchEventArgs) Handles MyBase.DatabaseVersionMismatch
            e.Updater.Update()
            e.Handled = True
        End Sub

        Private Sub InitializeComponent()
            Me.module1 = New DevExpress.ExpressApp.SystemModule.SystemModule()
            Me.module2 = New DevExpress.ExpressApp.Web.SystemModule.SystemAspNetModule()
            Me.module3 = New RemoveNewButtonInLookups.Module.RemoveNewButtonInLookupsModule()
            Me.module5 = New DevExpress.ExpressApp.Validation.ValidationModule()
            Me.module6 = New DevExpress.ExpressApp.Objects.BusinessClassLibraryCustomizationModule()
            Me.securityModule1 = New DevExpress.ExpressApp.Security.SecurityModule()
            CType(Me, System.ComponentModel.ISupportInitialize).BeginInit()
            ' 
            ' module5
            ' 
            Me.module5.AllowValidationDetailsAccess = True
            ' 
            ' RemoveNewButtonInLookupsAspNetApplication
            ' 
            Me.ApplicationName = "RemoveNewButtonInLookups"
            Me.Modules.Add(Me.module1)
            Me.Modules.Add(Me.module2)
            Me.Modules.Add(Me.module6)
            Me.Modules.Add(Me.module3)
            Me.Modules.Add(Me.module5)
            Me.Modules.Add(Me.securityModule1)
'            Me.DatabaseVersionMismatch += New System.EventHandler(Of DevExpress.ExpressApp.DatabaseVersionMismatchEventArgs)(Me.RemoveNewButtonInLookupsAspNetApplication_DatabaseVersionMismatch);
            CType(Me, System.ComponentModel.ISupportInitialize).EndInit()

        End Sub
    End Class
End Namespace
